libs/community/langchain_community/embeddings/huggingface.py

```diff
@@ -321,15 +340,30 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
                 texts = ["Hello, world!", "How are you?"]
                 hf_embeddings.embed_documents(texts)
         """  # noqa: E501
-        response = requests.post(
-            self._api_url,
-            headers=self._headers,
-            json={
-                "inputs": texts,
-                "options": {"wait_for_model": True, "use_cache": True},
-            },
-        )
-        return response.json()
+        all_embeddings = []
+        length_sorted_idx = np.argsort([-self._text_length(sen) for sen in texts])
+        sentences_sorted = [texts[idx] for idx in length_sorted_idx]
```

Comment:
where is this used?

Reply:
### Example Code

```
embeddings = HuggingFaceInferenceAPIEmbeddings(
    api_key=inference_api_key, 
    api_url=api_url,
    model_name="bge-large-en-v1.5"
)
pinecone.init(api_key=os.getenv("PINECONE_API_KEY"), environment=environment)

loader = PyPDFDirectoryLoader("data")
docs = loader.load()
text_splitter = RecursiveCharacterTextSplitter(chunk_size=2000, chunk_overlap=200)
chunks = text_splitter.split_documents(docs)

vectordb = Pinecone.from_documents(chunks, embeddings, index_name=index_name, namespace=namespace)
```
this code snippet is getting 314 request code from huggingface.py
```
response = requests.post(
            self._api_url,
            headers=self._headers,
            json={
                "inputs": texts,
                "options": {"wait_for_model": True, "use_cache": True},
            },
        )
        return response.json()
```
we should support batch size here. like local model embedding 

### Description

I am trying to use pinecone with hugging face inference for the embedding model. My total chunks are 420. and it is trying to process in one request.
Also embedding_chunk_size is not parsable from Pinecone.from_documents() method